libwasmvm/src/cache.rs

```diff
@@ -28,59 +27,21 @@ pub fn to_cache(ptr: *mut cache_t) -> Option<&'static mut Cache<GoApi, GoStorage
 
 #[no_mangle]
 pub extern "C" fn init_cache(
-    data_dir: ByteSliceView,
-    available_capabilities: ByteSliceView,
-    cache_size: u32,            // in MiB
-    instance_memory_limit: u32, // in MiB
+    config: ByteSliceView,
     error_msg: Option<&mut UnmanagedVector>,
 ) -> *mut cache_t {
-    let r = catch_unwind(|| {
-        do_init_cache(
-            data_dir,
-            available_capabilities,
-            cache_size,
-            instance_memory_limit,
-        )
-    })
-    .unwrap_or_else(|err| {
+    let r = catch_unwind(|| do_init_cache(config)).unwrap_or_else(|err| {
```

Comment:
Probably we don't need the `catch_unwind` here anymore - the `do_init_cache` doesn't seem to panic in any case

Reply:
None of these calls are supposed to panic. The `catch_unwind` is more like a safeguard in case we miss some case where the host can panic after all.